pkg/nfd-topology-updater/nfd-topology-updater.go

```diff
@@ -261,3 +264,21 @@ func createTopologyAttributes(policy string, scope string) v1alpha2.AttributeLis
 		},
 	}
 }
+
+func updateAttributes(lhs, rhs v1alpha2.AttributeList) v1alpha2.AttributeList {
+	check := make(map[string]string, len(lhs)+len(rhs))
+
+	for _, lattr := range lhs {
+		check[lattr.Name] = lattr.Value
+	}
+
+	for _, rattr := range rhs {
+		check[rattr.Name] = rattr.Value
+	}
+
+	ret := v1alpha2.AttributeList{}
+	for k, v := range check {
+		ret = append(ret, v1alpha2.AttributeInfo{Name: k, Value: v})
+	}
```

Comment:
I'm a bit concerned that with this any time any attribute is updated you basically get the list/slice re-shuffled in random order. Possibly ideally, we'd preserve the order of lhs with new attributes just appended.

Also, the current implementation is merely like `mergeAttributes()`. But should it be possible to delete attributes with this?

Any thoughts @ffromani and @PiotrProkop ?

Would be good to get this utility function into the NRT api...

Reply:
> So the question is do we want to make it work for more than one attribute from the start or just update podfingerprint for now.

Good question – I'm open to either approach. But if we have a generic function like this it should do the "right thing"

Reply:
my take is this code should only manage the attributes @PiotrProkop mentioned above, leaving everything else alone (= ignoring anything else). `nfd-topology-updater` should be the source of truth in the cluster for these attributes, so it should be allowed to add/update/overwrite/delete them, but just them.

Reply:
@ffromani sounds legit to me 👍

Reply:
I would say it is, when thinking about patching the object for example. But wdyt @ffromani @PiotrProkop ?

Reply:
I'm not against keeping attributes ordered if we can help it, because this minimizes the changes. IOW, we should unnecessarily change objects or their fields. However well-behaving, robust clients should NOT rely on any ordering and treat the sequence as unordered.

Reply:
> my take is this code should only manage the attributes @PiotrProkop mentioned above, leaving everything else alone (= ignoring anything else). `nfd-topology-updater` should be the source of truth in the cluster for these attributes, so it should be allowed to add/update/overwrite/delete them, but just them.

Right now this function only add the `ScanResponse` attributes a newly created NRT or the the readed one. 
`ScanResponse` only manage `podfingerprint.Attribute.
Sorry, I think I do not fully understand what the new function is suppose to do ... could you please elaborate a little bit? Thanks

Reply:
Clients should survive, I agree. But changing the order on every update breaks diffing, patching etc

Reply:
> Clients should survive, I agree. But changing the order on every update breaks diffing, patching etc

I think we are in agreement then. `nfd-topology-updater` should preserve the existing order of attributes and be smart about add/update/delete (note: I think delete is very unlikely). This is also easy to unit-test

Reply:
@marquiz I hear you about adding utilities to NRT API. I'll work on a PR, but I'm not sure I can make it quick enough to be used in this PR

Reply:
The simplest case is that this function iterates over all Attributes, looking for `podfingerprint.Attribute` if it's there check if checksum should be updated if so, update it in place. If it's not there append it to the end of the list.

Reply:
> The simplest case is that this function iterates over all Attributes, looking for `podfingerprint.Attribute` if it's there -> check if checksum should be updated if so, update it in place. If it's not there append it to the end of the list.

I can implement this approach here and change it to use @ffromani utility functions when available.

Reply:
works for me

Reply:
Sounds good

Reply:
Addressed. Let me know what you think

Reply:
In general I'd have squashed the fixing commit here. Codewise looks reasonnable.